flink-table/flink-table-planner/src/main/java/org/apache/flink/table/planner/plan/nodes/exec/stream/StreamExecGroupTableAggregate.java

```diff
@@ -120,16 +119,16 @@ protected Transformation<RowData> translateToPlanInternal(
 
         final AggsHandlerCodeGenerator generator =
                 new AggsHandlerCodeGenerator(
-                                new CodeGeneratorContext(
-                                        config, planner.getFlinkContext().getClassLoader()),
-                                planner.createRelBuilder(),
-                                JavaScalaConversionUtil.toScala(inputRowType.getChildren()),
-                                // TODO: heap state backend do not copy key currently,
-                                //  we have to copy input field
-                                // TODO: copy is not need when state backend is rocksdb,
-                                //  improve this in future
-                                // TODO: but other operators do not copy this input field.....
-                                true)
+                        new CodeGeneratorContext(
+                                config, planner.getFlinkContext().getClassLoader()),
+                        planner.createRelBuilder(),
+                        JavaScalaConversionUtil.toScala(inputRowType.getChildren()),
+                        // TODO: heap state backend do not copy key currently,
```

Comment:
I am not sure of the impact of these TODOs.

it would be worth tracking these TODOs with Jiras and including the numbers in the code.

Reply:
Currently, sync and asynchronous use different versions of state objects. In consideration of https://github.com/apache/flink/commit/bccace0f4b233e10279c7d95e009ae6aadad5ae8, the design concept is to design another set of corresponding files and reuse the code logic as much as possible in `Helper`